File: 874-backspace-string-compare/backspace-string-compare.dart
class Solution {
  bool backspaceCompare(String s, String t) {
    String ss = processString(s);
    String tt = processString(t);
    return ss == tt;
  }

  String processString(String str) {
    String result = '';
    for (int i = 0; i < str.length; i++) {
      if (str[i] == '#') {
        if (result.isNotEmpty) {
          result = result.substring(0, result.length - 1);
        }
      } else {
        result += str[i];
      }
    }
    return result;
  }
}

File: 874-backspace-string-compare/backspace-string-compare.cpp

class Solution {
public:
    bool backspaceCompare(string s, string t) {
        string ss = processString(s);
        string tt = processString(t);
        return ss == tt;
    }

private:
    string processString(string str) {
        string result;
        for (char ch : str) {
            if (ch == '#') {
                if (!result.empty()) {
                    result.pop_back();
                }
            } else {
                result.push_back(ch);
            }
        }
        return result;
    }
};